src/Orleans/Streams/PersistentStreams/PersistentStreamProvider.cs

```diff
@@ -60,11 +60,18 @@ public GrainFactoryServiceProvider(IStreamProviderRuntime providerRuntime)
             }
             public object GetService(Type serviceType)
             {
-                if (serviceType == typeof (GrainFactory))
+                var service = providerRuntime.ServiceProvider?.GetService(serviceType);
```

Comment:
Provider must have a runtime to work.  Null check here wouldn't help.  If we were to add a check we should check the value at provider initialization or at latest service provider creation.